execution/executor/src/components/chunk_output.rs

```diff
@@ -177,7 +178,31 @@ pub fn update_counters_for_processed_chunk(
                 .inc();
         }
 
+        if let Transaction::BlockMetadata(_) = txn {
+            match output.status() {
+                TransactionStatus::Discard(status) => {
```

Comment:
is Keep with success()=false also invalid outcome of the BlockMetadata transaction?

Reply:
Correct. Block prolog should always succeed.